haskell-src/exec/Chainweb/Server.hs

```diff
@@ -126,43 +127,53 @@ throttlingFactor load = if loadPerCap <= 1 then 1 else 1 / loadPerCap where
   -- without any slowdown
   loadPerCap = fromInteger load / 3
 
-apiServerCut :: Env -> ServerEnv -> ByteString -> IO ()
-apiServerCut env senv cutBS = do
+apiServerCut :: Env -> WorkerEnv -> ByteString -> IO ()
+apiServerCut env wenv cutBS = do
   let curHeight = cutMaxHeight cutBS
       logg = _env_logger env
   t <- getCurrentTime
   let circulatingCoins = getCirculatingCoins (fromIntegral curHeight) t
   logg Info $ fromString $ "Total coins in circulation: " <> show circulatingCoins
   let pool = _env_dbConnPool env
-  _ <- forkIO $ scheduledUpdates env pool (_serverEnv_runFill senv) (_serverEnv_fillDelay senv)
-  _ <- forkIO $ retryingListener env
-  logg Info $ fromString "Starting chainweb-data server"
-  throttledPool <- do
-    loadedSrc <- mkLoadedSource $ M.managed $ P.withResource pool
-    return $ do
-      loadedRes <- loadedSrc
-      load <- M.liftIO (lrLoadRef loadedRes)
-      return (lrResource loadedRes, throttlingFactor load)
-
-  let unThrottledPool = fst <$> throttledPool
-  let serverApp req =
-        ( ( recentTxsHandler logg unThrottledPool
-            :<|> searchTxs logg throttledPool req
-            :<|> evHandler logg throttledPool req
-            :<|> txHandler logg unThrottledPool
-            :<|> txsHandler logg unThrottledPool
-            :<|> accountHandler logg throttledPool req
-          )
-            :<|> statsHandler logg unThrottledPool
-            :<|> coinsHandler logg unThrottledPool
-          )
-          :<|> richlistHandler
-  let swaggerServer = swaggerSchemaUIServer Spec.spec
-      noSwaggerServer = throw404 "Swagger UI server is not enabled on this instance"
-  Network.Wai.Handler.Warp.run (_serverEnv_port senv) $ setCors $ \req f ->
-    if _serverEnv_serveSwaggerUi senv
-      then serve (Proxy @ApiWithSwaggerUI) (serverApp req :<|> swaggerServer) req f
-      else serve (Proxy @ApiWithNoSwaggerUI) (serverApp req :<|> noSwaggerServer) req f
+  let mETL = wenv ^? _Full . _2 <|> wenv ^? _ETL 
+  case mETL  of
+    Just (ETLEnv runFill fillDelay) -> do
+      _ <- forkIO $ scheduledUpdates env pool runFill fillDelay
+      _ <- forkIO $ retryingListener env
+      return ()
+    Nothing -> return ()
```

Comment:
Now that we've completely decoupled the ETL from the HTTP server, I think it makes more sense to start these threads outside `Server.hs`, at [this block](https://github.com/kadena-io/chainweb-data/blob/0382866dfc2ba8ac6985c9ca785aae7cb04ab9ac/haskell-src/exec/Main.hs#L85C1-L92) where we're dispatching to the `apiServer` function. Once we move this logic out of `apiServerCut`, we can make its type more precise by letting it accept `HTTPEnv` only.